src/api/environment.cc

```diff
@@ -587,10 +587,10 @@ bool InitializeContextForSnapshot(Local<Context> context) {
 
   context->SetEmbedderData(ContextEmbedderIndex::kAllowWasmCodeGeneration,
                            True(isolate));
-  return InitializePrimordials(context);
+  return !InitializePrimordials(context).IsEmpty();
```

Comment:
Can you move the `TODO` comment to here then, since this function still returns a `bool`?

Reply:
@addaleax I thought the `TODO` comment was meant only for the `InitializePrimordials()` function. I could also edit `InitializeContextForSnapshot()` and `InitializeContext()` to return `MaybeLocal`s if that is wanted since those are the only other functions up the call stack that return `bool`s. Also, I'm not entirely sure why we're supposed to do this, do you know why?

Reply:
> Also, I'm not entirely sure why we're supposed to do this, do you know why?

@RaisinTen Because `Maybe` and `MaybeLocal` are the way to signal to the caller that an exception is pending. And if `InitializePrimordials` here returns with an exception pending, then `InitializeContextForSnapshot` also returns with an exception pending, and consequently the return type of it should also be a `Maybe`.

> I could also edit `InitializeContextForSnapshot()` and `InitializeContext()` to return `MaybeLocal`s if that is wanted since those are the only other functions up the call stack that return `bool`s.

Yeah, this is probably the reason why the TODO comment was not addressed yet :) `InitializeContext(context)` is a public API, so changing the return type is semver-major. I think it’s probably still the best to do it.

(`InitializeContextRuntime` should also return a `Maybe<>`, fwiw).

Reply:
@addaleax Done